Completion: Fix completion through an import pointing to a local bind

julienduchesne opened the pull request:
Closes https://github.com/grafana/jsonnet-language-server/issues/113

This also fixes the go-to-definition functionality for this case, since it's the same code

Duologic commented:
This branch works with the case from #113 but fails on this one, a bit of an import chain:

```
// hello.jsonnet
local hello = {
  to: {
    the: 'world',
  },
};

hello
```

```
// hello2.jsonnet
local hello = import 'hello.jsonnet';

hello  // works
//hello.to // does not work
```

```
local g = import 'hello2.jsonnet';

{
  a: g,
}
```

julienduchesne commented:
> This branch works with the case from #113 but fails on this one, a bit of an import chain:

Thanks! I fixed the case where it's a short import chain or a longer one and I added some tests to check it. Should work with an infinite chain now

Duologic reviewed (approved):
works on my machine
